Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -0,0 +1,104 @@
+import libmongoc
+
+/// An enumeration of possible ReadConcern levels.
+public enum ReadConcernLevel: String {
+    case local
+    case available
+    case majority
+    case linearizable
+    case snapshot
+}
+
+/// A class to represent a MongoDB read concern.
+public class ReadConcern: Equatable, CustomStringConvertible {
+
+    /// A pointer to a mongoc_read_concern_t
+    internal var _readConcern: OpaquePointer?
+
+    /// The level of this readConcern, or nil if the level is not set.
+    public var level: String? {
+        guard let level = mongoc_read_concern_get_level(self._readConcern) else {
+            return nil
+        }
+        return String(cString: level)
+    }
+
+    private var asDocument: Document {
+        let doc = Document()
+        try? self.append(to: doc)
+        return doc
+    }
+
+    public var description: String {
+        return self.asDocument.description
+    }
+
+    /// Initialize a new ReadConcern from a ReadConcernLevel.
+    public convenience init(_ level: ReadConcernLevel) throws {
+        try self.init(level.rawValue)
+    }
+
+    /// Initialize a new ReadConcern from a String corresponding to a read concern level.
+    public init(_ level: String) throws {
```

Comment:
hmm, looking at the code in the tests below, I think we should stop these from throwing. I think this looks unintuitive to me:
```
let options1 = RunCommandOptions(readConcern: try ReadConcern(.local))`
```
As far as I've read the mongoc docs it looks like returning false will already emit a log error. As an alternative, I propose that we add an extra default level for a `ReadConcern`: `invalid`. In the case of an error, set the level to `invalid` (or it could just be nil as well now that I think about it). When the `ReadConcern` is eventually used in some I/O operation we should throw an error _there_

Reply:
yeah I did feel like that was kind of odd myself as I wrote the tests. 

now that I look at `mongoc_read_concern_set_level`'s [source](https://github.com/mongodb/mongo-c-driver/blob/865e11a3498240dbd927e537dc3f212049f901fb/src/mongoc/mongoc-read-concern.c#L88), I don't think it ever actually returns `false`? from the [docs](http://mongoc.org/libmongoc/current/mongoc_read_concern_set_level.html) it looks like that used to happen if the ReadConcern was "frozen" which no longer happens as of v1.9. I think as long as we pass in a string (which we always will) it should return true. and if the string does not represent a valid RC, mongoc doesn't throw til we try to use it in an operation. 

so I think we can just ignore the result of this function, so let's get rid of the error. we could alternatively use a `precondition` to check that it's true but that's maybe overkill